app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
+        int i = 0;
         for (String key : sEmojiConverterMap.keySet()) {
-            replaceEditable(input, key, sEmojiConverterMap.get(key));
+            if (replaceEditable(input, key, sEmojiConverterMap.get(key))){
+                i++;
```

Comment:
Yeah, I see your points. A bool would also do it. But does it has an influence on the performance? Maybe it takes less space...